src/client/datascience/jupyter/kernels/helpers.ts

```diff
@@ -361,6 +361,8 @@ export function findPreferredKernel(
         if (kernelMatchingPreferredInterpreter) {
             return kernelMatchingPreferredInterpreter;
         }
+        // Telemetry to see if this happens in the real world, this should not be possible.
+        sendTelemetryEvent(Telemetry.FailedToFindKernelSpecInterpreterForInteractive);
```

Comment:
Added telemetry to understand if users run into this. Looking at the code, it doesn't make sense how this case would hit